src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -168,13 +172,31 @@ public function load(array $configs, ContainerBuilder $container)
         $container->getDefinition('security.authentication.guard_handler')
             ->replaceArgument(2, $this->statelessFirewallKeys);
 
+        // @deprecated since Symfony 5.3
         if ($config['encoders']) {
             $this->createEncoders($config['encoders'], $container);
+            $config['password_hashers'] = $config['encoders'];
```

Comment:
this assignment means that if you explicitly configure both, the new non-deprecated config is silently lost. This looks wrong to me.

Reply:
if we want to consider both configs equivalent (as suggested below), my suggestion would actually be perform this rewriting on a per-file basis in the Configuration class (using `beforeNormalization()`, not `validate()`) so that the usage of the old API in one of the file does not make you loose the config done in the modern way in the other file.